src/core/registration.c

```diff
@@ -58,39 +54,15 @@ MsQuicRegistrationOpen(
         goto Error;
     }
 
-    CXPLAT_DBG_ASSERT(ExternalRegistration || MsQuicLib.Datapath != NULL);
-
     if (ExternalRegistration) {
-        CxPlatLockAcquire(&MsQuicLib.Lock);
-        if (MsQuicLib.Datapath == NULL) {
-            CXPLAT_DATAPATH_CONFIG DataPathConfig = {
-                MsQuicLib.DataPathProcList,
-                MsQuicLib.DataPathProcListLength
-            };
-            Status =
-                CxPlatDataPathInitialize(
-                    sizeof(CXPLAT_RECV_PACKET),
-                    &DatapathCallbacks,
-                    NULL,                   // TcpCallbacks
-                    &DataPathConfig,
-                    &MsQuicLib.Datapath);
-            if (QUIC_FAILED(Status)) {
-                CxPlatLockRelease(&MsQuicLib.Lock);
-                QuicTraceEvent(
-                    LibraryErrorStatus,
-                    "[ lib] ERROR, %u, %s.",
-                    Status,
-                    "CxPlatDataPathInitialize");
-                goto Error;
-            }
-            QuicTraceEvent(
-                DataPathInitialized,
-                "[data] Initialized, DatapathFeatures=%u",
-                CxPlatDataPathGetSupportedFeatures(MsQuicLib.Datapath));
```

Comment:
This refactor seems unnecessary to me. QuicLibraryEnsureExecutionContext is essentially just initialize datapath.

Reply:
As it is now, I'd agree with you. But in an effort to break this PR up some, I haven't made the QUIC layer changes to only run on the configured CPUs. That's the next step. With that work, a lot more will change in library.c around PartitionCount, so more will be added then.